lib/beaker-puppet/install_utils/puppet_utils.rb

```diff
@@ -90,41 +90,51 @@ def get_puppet_collection(agent_version = 'latest')
           collection
         end
 
-        # Determine the puppet collection that matches a given version of the puppet-agent
-        # package (you can find this version in the `aio_agent_version` fact).
+        # Determine the puppet collection that matches a given package version. The package
+        # must be one of
+        #   * :puppet_agent (you can get this version from the `aio_agent_version_fact`)
+        #   * :puppet
         #
-        # @param agent_version [String] a semver version number of the puppet-agent package, or the string 'latest'
+        #
+        # @param package [Symbol] the package name. must be one of :puppet_agent, :puppet
+        # @param version [String] a version number or the string 'latest'
         # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_agent_version(agent_version)
-          agent_version = agent_version.to_s
-          return 'puppet' if agent_version.strip == 'latest'
+        def puppet_collection_for(package, version)
+          valid_packages = [
+            :puppet_agent,
+            :puppet
+          ]
 
-          x, y, z = agent_version.to_s.split('.').map(&:to_i)
-          return nil if x.nil? || y.nil? || z.nil?
+          unless valid_packages.include?(package)
+            raise "package must be one of #{valid_packages.join(', ')}"
+          end
 
-          return 'pc1' if x == 1
+          case package
+          when :puppet_agent
+            agent_version = version.to_s
+            return 'puppet' if agent_version.strip == 'latest'
 
-          # A y version >= 99 indicates a pre-release version of the next x release
-          x += 1 if y >= 99
-          "puppet#{x}" if x > 4
-        end
+            x, y, z = agent_version.to_s.split('.').map(&:to_i)
+            return nil if x.nil? || y.nil? || z.nil?
```

Comment:
Is there a way we can reduce the redundancies here? The only real difference between `:puppet_agent` and `:puppet` that I can tell is the pc1 logic (`return 'pc1' if x == 1` vs `return 'pc1' if x == 4`).

Reply:
Yeah, I can go ahead and do that.

Reply:
Updated the PR.